andela/hc-the-knights
# #153728037 Add multiple alert methods

#### What does this PR do?
Setting up healthchecks to be able to send out important information concerning the cron jobs through multiple channels

#### Description of Task to be completed?
Integration of several alert methods in the system such that the system is able to send a user alerts of all those selected platforms when a cron job is pinged and when it goes down.

#### How should this be manually tested?
Open health checks.io on your localhost
Navigate to the integrations page. 
* For email, click on the add integrations button in the email row, add the email you would like the notifications to be sent to. An email is sent to that email inbox to verify the setup. Click on the button in the email to confirm set up.

* For sms, click on the add integrations button in the sms row. You will be redirected to a page where you can add the phone number you would like to get your notifications on. Add the number and click on the save integration button. The number should be registered with twilio for it to work.

* For slack, click on the add integrations button in the sms row,  you are redirected to a page where you can select the channel to which the notifications will be sent to from your list of available channels 

#### The files changed/created include:
Created: 
add_sms.html - Added UI for the page where the phone number is added
.env - Added the twilio configuration settings

Edited
hc/settings.py - Added the smtp server settings and the email details
hc/local_settings.py - Added the SLACK configurations
hc/api/transports.py - Added the SMS model
hc/front/urls.py - Added the URL that redirects to the sms configuration page
hc/front/views.py - Added the function that loads the page where the number to sms is added
hc/templates/front/channels.html - Added sms to the list of configurable channel options
hc/front/tests/test_add_channel.py 

#### What are the relevant pivotal tracker stories?
[#153728037](https://www.pivotaltracker.com/story/show/153728037)

Screenshots
<img width="1150" alt="the channels" src="https://user-images.githubusercontent.com/5388763/35122784-ff2130d6-fcb0-11e7-90b1-89c733ac8699.png">
channels

<img width="1157" alt="cron jobs in various states" src="https://user-images.githubusercontent.com/5388763/35122812-166d4d42-fcb1-11e7-84ad-5636e1cdbe76.png">
Cron jobs in different states

<img width="949" alt="screen shot 2018-01-18 at 23 38 34" src="https://user-images.githubusercontent.com/5388763/35122871-4fb3c2d4-fcb1-11e7-8706-9b9ee9c2f963.png">
Email notifications

<img width="914" alt="slack notifications" src="https://user-images.githubusercontent.com/5388763/35122900-6c2a47a8-fcb1-11e7-9362-4b08ed4a0164.png">
Slack notifications

![sms notifications](https://user-images.githubusercontent.com/5388763/35122957-91b496c2-fcb1-11e7-8d52-93127c7b18b9.png)
SMS notifications
